### ISSUE TITLE
@neutrinojs/fork error while running:  WebpackOptionsValidationError: Invalid configuration object

### ISSUE THREAD

**Issue by hashtd** [mozilla-neutrino/neutrino-dev#588]
getting following error when using `@neutrinojs/fork` (8.0.9) for react & node

```
> nn_fork@1.0.0 start /home/ubuntu/projects/nn_fork
> neutrino start

✖ Building project failed
WebpackOptionsValidationError: Invalid configuration object. Webpack has been initialised using a configuration object that does not match the API schema.
 - configuration misses the property 'entry'.
   object { <key>: non-empty string | [non-empty string] } | non-empty string | [non-empty string] | function
   -> The entry point(s) of the compilation.
    at ChainAction.validator.chain.errors [as mapper] (/home/ubuntu/projects/nn_fork/node_modules/neutrino/src/webpack.js:49:20)
    at ChainAction$resolved [as resolved] (/home/ubuntu/projects/nn_fork/node_modules/fluture/index.js:798:27)
    at resolved (/home/ubuntu/projects/nn_fork/node_modules/fluture/index.js:213:19)
    at Resolved._fork (/home/ubuntu/projects/nn_fork/node_modules/fluture/index.js:636:3)
    at drain (/home/ubuntu/projects/nn_fork/node_modules/fluture/index.js:270:25)
    at settle (/home/ubuntu/projects/nn_fork/node_modules/fluture/index.js:199:15)
    at resolved (/home/ubuntu/projects/nn_fork/node_modules/fluture/index.js:213:5)
    at Immediate.EncaseP2$res (/home/ubuntu/projects/nn_fork/node_modules/fluture/index.js:1840:7)
    at runCallback (timers.js:791:20)
    at tryOnImmediate (timers.js:751:5)
npm ERR! code ELIFECYCLE
npm ERR! errno 1
npm ERR! nn_fork@1.0.0 start: `neutrino start`
npm ERR! Exit status 1
npm ERR! 
npm ERR! Failed at the nn_fork@1.0.0 start script.
npm ERR! This is probably not a problem with npm. There is likely additional logging output above.

npm ERR! A complete log of this run can be found in:
npm ERR!     /home/ubuntu/.npm/_logs/2017-12-10T07_47_12_696Z-debug.log

```
.neutrinorc.js
```
module.exports = {
  options: {},
  use: [
    [
      '@neutrinojs/fork', {
        configs: {
          react: '@neutrinojs/react',
          node: {
            options: { source: 'server', output: 'build/server' },
            use: ['@neutrinojs/node']
          }
        }
      }
    ]
  ]
};

```
on ubuntu 17.10, nodejs v8.9.3, npm 5.6.0, neutrinojs 8.0.9

**Comment by hashtd** [comment]
after inspecting found that in `base.js` getting error middleware.forEach is not function
`config` at `start.js` is coming empty

now after making following changes its now working

```
diff --git a/packages/neutrino/bin/base.js b/packages/neutrino/bin/base.js
index 5e6dbf3..a99efeb 100644
--- a/packages/neutrino/bin/base.js
+++ b/packages/neutrino/bin/base.js
@@ -22,7 +22,11 @@ module.exports = ({
   }, args.options);
   const api = Neutrino(options);
 
-  middleware.forEach(middleware => api.use(middleware));
+  if (typeof(middleware) == 'function') {
+    middleware.forEach(middleware => api.use(middleware));
+  } else {
+    api.use(middleware)
+  }
 
   if (!apiCommand || args.help) {
     Object
diff --git a/packages/neutrino/bin/start.js b/packages/neutrino/bin/start.js
index 1cfc936..4daf789 100644
--- a/packages/neutrino/bin/start.js
+++ b/packages/neutrino/bin/start.js
@@ -1,4 +1,5 @@
 const ora = require('ora');
+const { isEmpty } = require('ramda');
 const { start } = require('../src');
 const base = require('./base');
 
@@ -11,12 +12,14 @@ module.exports = (middleware, args, cli) => {
     args,
     NODE_ENV: 'development',
     commandHandler(config, neutrino) {
-      if (!args.start) {
-        spinner.enabled = global.interactive;
-        spinner.start();
-      }
+      if(!isEmpty(config)) {
+        if (!args.start) {
+          spinner.enabled = global.interactive;
+          spinner.start();
+        }
 
-      return start(config, neutrino);
+        return start(config, neutrino);
+      }
     },
     errorsHandler() {
       spinner.fail('Building project failed');
```

**Comment by eliperelman** [comment]
Thanks for the issue! Sorry about this, I have a patch in flight that will fix this, and I will get that published as soon as the tests pass.

**Comment by eliperelman** [comment]
This is now released in v8.0.10. Give it a shot!
